### ISSUE TITLE
Failure during test-bot's brew pull: `exec': no implicit conversion of MatchData into String (TypeError)

### ISSUE THREAD

**Issue by scpeters** [Homebrew/homebrew-test-bot#157]
I have some bottle builds that are failing with an error that I can't figure out (I've removed `/usr/local/Homebrew/Library/Taps/homebrew/homebrew-test-bot/` from the console log to improve readability):

~~~
+ brew test-bot --tap=osrf/simulation --ci-pr https://github.com/osrf/homebrew-simulation/pull/418
Homebrew/homebrew-test-bot ed0844f (Merge pull request #156 from MikeMcQuaid/test-bot-new-formula-pr-url)
ARGV: --tap=osrf/simulation --ci-pr https://github.com/osrf/homebrew-simulation/pull/418 --cleanup --test-default-formula --local --junit
==> Tapping osrf/simulation
Cloning into '/usr/local/Homebrew/Library/Taps/osrf/homebrew-simulation'...
Tapped 48 formulae (126 files, 745.5KB)
==> git -C Homebrew checkout master -f
==> git -C Homebrew reset --hard origin/master
==> brew pull --clean https://github.com/osrf/homebrew-simulation/pull/418
==> FAILED
cmd/brew-test-bot.rb:301:in `exec': no implicit conversion of MatchData into String (TypeError)
	from cmd/brew-test-bot.rb:301:in `block in run'
	from cmd/brew-test-bot.rb:296:in `fork'
	from cmd/brew-test-bot.rb:296:in `run'
	from cmd/brew-test-bot.rb:1185:in `test'
	from cmd/brew-test-bot.rb:485:in `download'
	from cmd/brew-test-bot.rb:1253:in `run'
	from cmd/brew-test-bot.rb:1626:in `block in test_bot'
	from cmd/brew-test-bot.rb:1609:in `each'
	from cmd/brew-test-bot.rb:1609:in `test_bot'
	from cmd/brew-test-bot.rb:1701:in `<top (required)>'
	from /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.3.3_2/lib/ruby/2.3.0/rubygems/core_ext/kernel_require.rb:55:in `require'
	from /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.3.3_2/lib/ruby/2.3.0/rubygems/core_ext/kernel_require.rb:55:in `require'
	from /usr/local/Homebrew/Library/Homebrew/utils.rb:18:in `require?'
	from /usr/local/Homebrew/Library/Homebrew/brew.rb:101:in `<main>'
==> brew cleanup --prune=7
Error: undefined method `gsub' for #<MatchData:0x000001019e8390>
cmd/brew-test-bot.rb:488:in `download'
cmd/brew-test-bot.rb:1253:in `run'
cmd/brew-test-bot.rb:1626:in `block in test_bot'
cmd/brew-test-bot.rb:1609:in `each'
cmd/brew-test-bot.rb:1609:in `test_bot'
cmd/brew-test-bot.rb:1701:in `<top (required)>'
/usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.3.3_2/lib/ruby/2.3.0/rubygems/core_ext/kernel_require.rb:55:in `require'
/usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.3.3_2/lib/ruby/2.3.0/rubygems/core_ext/kernel_require.rb:55:in `require'
/usr/local/Homebrew/Library/Homebrew/utils.rb:18:in `require?'
/usr/local/Homebrew/Library/Homebrew/brew.rb:101:in `<main>'
~~~

* [high sierra failure](https://build.osrfoundation.org/job/generic-release-homebrew_bottle_builder-highsierra/164/console)
* [el capitan failure](https://build.osrfoundation.org/job/generic-release-homebrew_bottle_builder-elcapitan/226/console)

I can reproduce this interactively on my build machines. `brew doctor` is clean:

~~~
$ brew doctor
Your system is ready to brew.
$ brew config
HOMEBREW_VERSION: 1.6.4-24-g24025a3
ORIGIN: https://github.com/Homebrew/brew
HEAD: 24025a3ab52822bd8395db47a61b73e45940cc28
Last commit: 5 hours ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: e8561625765c1a3100a04ad8c73ada213f8c2f77
Core tap last commit: 5 hours ago
HOMEBREW_PREFIX: /usr/local
HOMEBREW_DEV_CMD_RUN: 1
CPU: octa-core 64-bit ivybridge
Homebrew Ruby: 2.3.3 => /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.3.3_2/bin/ruby
Clang: 8.0 build 800
Git: 2.10.1 => /Applications/Xcode.app/Contents/Developer/usr/bin/git
Curl: 7.43.0 => /usr/bin/curl
Java: 1.8.0_131, 1.8.0_77
macOS: 10.11.6-x86_64
CLT: 8.2.0.0.1.1480973914
Xcode: 8.2.1
XQuartz: 2.7.11 => /opt/X11
~~~

I can execute `brew pull --clean https://github.com/osrf/homebrew-simulation/pull/418` from the command-line but not through `test-bot`. Our last successful bottle builds were a week ago. Any ideas?

**Comment by scpeters** [comment]
Reverting #156 fixes this for me

**Comment by scpeters** [comment]
Fix is in #158
